### DISCUSSION TITLE
Does Renovate support dynamically capturing assignee from my own JSON file?

### DISCUSSION THREAD

**Discussion by Devinwong** [renovatebot/renovate#31056] [Request Help] [answered]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Mend Renovate hosted app on github.com

### Please tell us more about your question or problem

Hi Renovate team. I know that we can assign a PR based on a specific package to a specific person by using `packageRules` like
```
   "packageRules": [
    {
      "matchPackageNames": ["packageA"],
      "assignees": ["personA"]
    },
    {
      "matchPackageNames": ["packageB"],
      "assignees": ["personB"]
    }
  ],
```

I am wondering if Renovate supports capturing the assignee dynamically from my own JSON file (e.g., mypackages.json), instead of statically defined in Renovate.json? The benefit of this request is to keep my renovate.json short and abstracted.

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by rarkins** [comment, marked as answer]
No, this is not supported

**Reply by Devinwong** [reply]
Thanks for confirming, though

**Comment by secustor** [comment]
You can do this via [HTTP hosted presets](https://docs.renovatebot.com/config-presets/#fetching-presets-from-an-http-server), that is if you can provide the presets in the expected format.

**Reply by Devinwong** [reply]
Secustor, thanks for your suggestion too.
I don't really understand presets yet.
From this doc, https://docs.renovatebot.com/config-presets/, it has
```
{
  "labels": {
    "description": "Apply labels <code>{{arg0}}</code> and <code>{{arg1}}</code> to PRs",
    "labels": ["{{arg0}}", "{{arg1}}"]
  },
  "assignee": {
    "description": "Assign PRs to <code>{{arg0}}</code>",
    "assignees": ["{{arg0}}"]
  }
}
```
If I can change the {{arg0}} to a regex that can read from my json file, then it will achieve my goal.
Do you know if that's possible?

**Reply by secustor** [reply]
This is explained directly below that example. That is for usage in `extends` blocks.

What I mean is that you can generate a preset JSON with a method you are seeing fit, host it somewhere and let Renovate that preset to achieve your goal.

**Reply by Devinwong** [reply]
Oh thanks. Will look into it for more details.
